### PR TITLE
Update create-mesh to support off-GKE clusters

### PR DESCRIPTION
**Pull request by backward-compatible**

### CODE REVIEW

File: asmcli/commands/create-mesh.sh [hunk 1]
```diff
@@ -70,105 +76,76 @@ create-mesh_validate_args() {
...
-      if [[ "${CTX_CLUSTER1}" != "${CTX_CLUSTER2}" ]]; then
-        install_one_remote_secret "${CTX_CLUSTER1}" "${CTX_CLUSTER2}"
+  while read -r KCF1; do
+    while read -r KCF2; do
+      if [[ "${KCF1}" != "${KCF2}" ]]; then
+        install_one_remote_secret "${KCF1}" "${KCF2}"
       fi
     done <<EOF
-$(context_list "clusterContexts")
+$(m1context_list "kubeconfigFiles")
```
Comment (zerobfd) [inline asmcli/commands/create-mesh.sh:131]: What's the `m1` here?

File: asmcli/lib/config.sh [hunk 2]
```diff
@@ -66,14 +66,15 @@ configure_package() {
...
-  local CLUSTER_NAME; CLUSTER_NAME="${3}"
   local CONTEXT; CONTEXT="$(context_get-option "CONTEXT")"
   local KUBECONFIG; KUBECONFIG="$(context_get-option "KUBECONFIG")"
   local KUBECONFIG_SUPPLIED; KUBECONFIG_SUPPLIED="$(context_get-option "KUBECONFIG_SUPPLIED")"
 
   if [[ "${KUBECONFIG_SUPPLIED}" -eq 0 ]]; then
+    local PROJECT_ID; PROJECT_ID="${1}"
+    local CLUSTER_LOCATION; CLUSTER_LOCATION="${2}"
+    local CLUSTER_NAME; CLUSTER_NAME="${3}"
+
```
Comment (zerobfd) [inline asmcli/lib/config.sh:77]: Was the intention here to be able to use `configure_kubectl` without passing in arguments?

File: asmcli/commands/create-mesh.sh [hunk 3]
```diff
@@ -1,11 +1,17 @@
 create-mesh_subcommand() {
   ### Preparation ###
+
+  # using kubeconfig globally for create-mesh sub-command
+  context_set-option "KUBECONFIG_SUPPLIED" 1
```
Comment (zerobfd) [inline asmcli/commands/create-mesh.sh:5]: I'd like to either change the name of this context variable, or change the logic that depends on it.

File: asmcli/commands/create-mesh.sh [hunk 4]
```diff
@@ -1,11 +1,17 @@
...
+  # setting same behavior for all environments
+  context_set-option "PLATFORM" "multicloud"
```
Comment (zerobfd) [inline asmcli/commands/create-mesh.sh:7]: Same as above

File: asmcli/commands/create-mesh.sh [hunk 5]
```diff
@@ -70,105 +76,76 @@ create-mesh_validate_args() {
...
 
 install_all_remote_secrets() {
-  local CTX_CLUSTER1 CTX_CLUSTER2
-
-  while read -r CTX_CLUSTER1; do
-    while read -r CTX_CLUSTER2; do
-      if [[ "${CTX_CLUSTER1}" != "${CTX_CLUSTER2}" ]]; then
-        install_one_remote_secret "${CTX_CLUSTER1}" "${CTX_CLUSTER2}"
+  while read -r KCF1; do
+    while read -r KCF2; do
```
Comment (zerobfd) [inline asmcli/commands/create-mesh.sh:126]: I like the shorter variable names, but could we pull the no-logic-change refactors into another PR that I can stamp and then rebase this PR on that commit? It makes it a bit difficult to review just the logic changes.

**Review by zerobfd — CHANGES_REQUESTED** [review]
The overall logic flow makes sense, just a few discrete issues.

File: asmcli/commands/create-mesh.sh (hunk 1 above)
Comment (backward-compatible) [reply asmcli/commands/create-mesh.sh:131]: sorry that's remnant from my local tests..

File: asmcli/lib/config.sh (hunk 2 above)
Comment (backward-compatible) [reply asmcli/lib/config.sh:77]: yes

File: asmcli/commands/create-mesh.sh (hunk 3 above)
Comment (backward-compatible) [reply asmcli/commands/create-mesh.sh:5]: why/to what?

File: asmcli/commands/create-mesh.sh (hunk 4 above)
Comment (backward-compatible) [reply asmcli/commands/create-mesh.sh:7]: i'm going to move the PLATFORM set call more locally but anyway why/to what would you like it to be renamed?

File: asmcli/commands/create-mesh.sh (hunk 5 above)
Comment (backward-compatible) [reply asmcli/commands/create-mesh.sh:126]: this specific one is more than just renaming, i'm changing the usage from **context** to **kubeconfig** so had rename accordingly.

File: asmcli/lib/config.sh (hunk 2 above)
Comment (backward-compatible) [reply asmcli/lib/config.sh:77]: looking at the code as it is rn, i feel like this is no longer needed

**Comment by backward-compatible** [comment]
merge conflict need to create a new PR

**Pull request closed without merging**